app.py

```diff
@@ -5,41 +5,78 @@
 from django.db import transaction  # type: ignore
 from openstates.cli.reports import generate_session_report
 import urllib.parse
+import json
 
 logger = logging.getLogger("openstates")
 s3_client = boto3.client("s3")
 s3_resource = boto3.resource("s3")
 
 
-def archive_processed_file(bucket, key):
+def archive_processed_file(bucket, all_keys):
     """
     Archive the processed file to avoid possible scenarios of race conditions.
     We currently use meta.client.copy instead of client.copy b/c it can copy
     multiple files via multiple threads, since we have batching in view.
 
     Args:
         bucket (str): The s3 bucket name
-        key (str): The key of the file to be archived
+        all_keys (list): The key of the file to be archived
     Returns:
         None
 
     Example:
         >>> archive_processed_file("my-bucket", "my-file.json")
     """
 
-    global s3_resource
+    for key in all_keys:
+        copy_source = {"Bucket": bucket, "Key": key}
 
-    copy_source = {"Bucket": bucket, "Key": key}
+        s3_resource.meta.client.copy(copy_source, bucket, f"archive/{key}")
+        logger.info(f"Archived file :: {key}")
 
-    s3_resource.meta.client.copy(copy_source, bucket, f"archive/{key}")
-    logger.info(f"Archived file :: {key}")
+        # delete object from original bucket
+        s3_client.delete_object(Bucket=bucket, Key=key)
+        logger.info(f"Deleted file :: {key}")
 
-    # delete object from original bucket
-    s3_client.delete_object(Bucket=bucket, Key=key)
-    logger.info(f"Deleted file :: {key}")
 
+def retrieve_messages_from_queue():
+    """
+    Get the file paths from the SQS.
+    """
+
+    # Create SQS client
+    sqs = boto3.client("sqs")
+
+    sqs_url = os.environ.get("SQS_QUEUE_URL")
+
+    # Receive message from SQS queue
+    response = sqs.receive_message(
+        QueueUrl=sqs_url,
+        AttributeNames=["SentTimestamp"],
+        MaxNumberOfMessages=10,
+        MessageAttributeNames=["All"],
+        VisibilityTimeout=0,
+        WaitTimeSeconds=0,
+    )
+
+    messages = response.get("Messages", [])
+    message_bodies = []
+
+    if messages:
+
+        for message in messages:
+            message_bodies.append(message.get("Body"))
+
+            receipt_handle = message["ReceiptHandle"]
+
+            # Delete received message from queue
+            sqs.delete_message(QueueUrl=sqs_url, ReceiptHandle=receipt_handle)
```

Comment:
Would it make any sense to avoid deleting the SQS message until processing has been completed successfully? what if there is some program error in the middle - could all SQS messages get consumed, and then it becomes impossible to restart/retry them without re-running the original scrape job?

Reply:
That makes sense. Although for unsuccessful runs we will see them in s3 too, I've added a todo for this.